lib/postgrex/connection.ex

```diff
@@ -85,16 +85,46 @@ defmodule Postgrex.Connection do
     message = {:query, statement, params}
     timeout = opts[:timeout] || @timeout
     case Connection.call(pid, message, timeout) do
-      %Postgrex.Result{} = res ->
-        {:ok, res}
-      %Postgrex.Error{} = err ->
-        {:error, err}
       {:exit, reason} ->
         exit({reason, {__MODULE__, :query, [pid, statement, params, opts]}})
+      result ->
+        result
```

Comment:
Will need this change in `listen`/`unlisten` to as the `%Postgrex.Error{}` can will now be sent as `{:error, %Postgrex.Error{}}`. There should be a test for these if its possible.